doc/reference/serialization.rst

```diff
@@ -0,0 +1,267 @@
+Message serialization
+=====================
+
+IPv8 gives you as much control as possible over the messages you send over the Internet.
+The ``Overlay`` (or ``Community``) class lets you send arbitrary strings over the (UDP) ``endpoint``.
+However efficient this may be, having non-standardized string contruction for each message of your overlay can distract from the overal overlay design.
+This is the age-old dichotomy of maintainable versus performant code.
+
+The basic class for serializing objects from and to strings/network packets is the ``Serializer`` (``ipv8/messaging/serialization.py``).
+Though the ``Serializer`` is extensible, you will mostly only need the default serializer ``default_serializer``.
+You can use the ``Serializer`` with classes of the following types:
+
+.. csv-table:: Serializable classes
+   :header: "class", "path", "description"
+   :widths: 10, 10, 20
+
+   "Serializable", "ipv8/messaging/serialization.py", "Base class for all things serializable. Should support the instance method to_pack_list() and the class method from_unpack_list()."
+   "Payload", "ipv8/messaging/payload.py", "Extension of the Serializable class with logic for pretty printing."
+   "VariablePayload", "ipv8/messaging/lazy_payload.py", "Less verbose way to specify Payloads, at the cost of performance."
+
+
+Each of these serializable classes specifies a list of primitive data types it will serialize to and from.
+The primitive data types are specified in the :ref:`data types<Datatypes Section>` Section.
+Each serializable class has to specify the following class members:
+
+.. csv-table:: Serializable class members
+   :header: "member", "description"
+   :widths: 10, 20
+
+   "format_list", "A list containing valid data type primitive names."
+   "optional_format_list", "A list containing valid data type primitive names. As the name implies, any number of arguments may be supplied."
+   "names", "Only for VariablePayload classes, the instance fields to bind the data types to."
+
+
+As an example, we will now define three completely wire-format compatible messages using the three classes.
+Each of the messages will serialize to a (four byte) unsigned integer followed by an optional (two byte) unsigned short.
+Each instance will have two fields: ``field1`` and ``field2`` corresponding to the integer and short.
+
+.. code-block:: python
+
+    class MySerializable(Serializable):
```

Comment:
Could you maybe provide an example with a string or raw field as well? The reader will probably like that.